dask_cuda/tests/test_explicit_comms.py

```diff
@@ -141,15 +147,13 @@ def _test_dataframe_shuffle(backend, protocol, n_workers, _partitions):
 
             for input_nparts in range(1, 5):
                 for output_nparts in range(1, 5):
-                    ddf = dd.from_pandas(df.copy(), npartitions=input_nparts).persist(
-                        workers=all_workers
-                    )
+                    ddf1 = dd.from_pandas(df.copy(), npartitions=input_nparts)
```

Comment:
**Strange Finding**: Even without the changes in this PR, I get a "cancelled" error when I persist `ddf` and then perform **both** an explicit comms shuffle and then a task-based shuffle. I don't understand the cause of this yet.

Reply:
Are we ok still moving ahead with the changes even though there's a cancelation error?

Reply:
Yes. The changes in this PR are independent of the cancellation error (it happens in `branch-24.08` without these changes).

Another important detail: I only get the error when query-planning is disabled. Therefore, I assume the problem has to do with a key-name collision that doesn't happen with dask-expr (which is much more disciplined about key names than the legacy API is).